### PR TITLE
[fastlane] Update error message when using fastlane_version with bundle

### LINKED ISSUE #6828: Inappropriate message when using fastlane_version action with bundler
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

When using `fastlane_version` action in a bundler environment (with `bundle exec fastlane`), the update message is not appropriate: 

```
$ bundle exec fastlane my_lane_farmer
-------------------------------------------------
--- Step: Verifying required fastlane version ---
-------------------------------------------------
[!] The Fastfile requires a fastlane version of >= 8.8.8. You are on 1.101.0. Please update using `sudo gem update fastlane`
```

It should ask the user to do a `bundle update fastlane`.

### PR DESCRIPTION
**Pull request by jlnquere**
Here is a PR to fix #6828 :)

Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes:

- [x] Run `bundle exec rspec` from the subdirectory of each tool you modified. Alternatively, run `rake test_all` from the root directory.
- [x] Run `bundle exec rubocop -a` to ensure the code style is valid
- [x] Read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] We currently don't accept new actions, please publish a plugin instead, more information in [Plugins.md](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Plugins.md)

Before submitting a pull request, we appreciate if you create an issue first to discuss the change :+1:

### CODE REVIEW

**Review by milch — CHANGES_REQUESTED** [review]
Looks good, except for the commit history: there's merge commits, duplicate commits, reverts, ... 

Please fix the history, e.g. by cherry picking your changes on top of the current master and force-pushing to this branch to get rid of the old history, then we can merge this.

**Comment by Onejjy** [comment]
Indeed, it was a mess 🙈. It should be better now !

File: fastlane/spec/actions_specs/fastlane_version_spec.rb [hunk 1]
```diff
@@ -12,9 +12,29 @@
           Fastlane::FastFile.new.parse("lane :test do
             fastlane_version '9999'
           end").runner.execute(:test)
+        end.to raise_error
+      end
+
+      it "raises an exception if it's an old version in a non-bundler environement" do
```
Comment (KrauseFx) [inline fastlane/spec/actions_specs/fastlane_version_spec.rb]: tests ❤️

File: fastlane/lib/fastlane/actions/fastlane_version.rb [hunk 2]
```diff
@@ -12,7 +12,13 @@ def self.run(params)
         UI.user_error!("Please pass minimum fastlane version as parameter to fastlane_version") unless defined_version
 
         if Gem::Version.new(Fastlane::VERSION) < defined_version
-          UI.user_error!("The Fastfile requires a fastlane version of >= #{defined_version}. You are on #{Fastlane::VERSION}. Please update using `sudo gem update fastlane`.")
+          update_message = nil
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/fastlane_version.rb]: You don't need to init this to `nil`

**Comment by milch** [comment]
@Onejjy Thanks for fixing the history! Once you fix Felix's suggestion we'll merge.

**Comment by Onejjy** [comment]
This is fixed :)

**Review by milch — APPROVED** [review]

**Pull request merged**